lib/matplotlib/scale.py

```diff
@@ -478,10 +478,111 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    a = a.copy()
```

Comment:
It looks like you switched in midstream from using masked arrays to using nans.  Does this need to handle masked array inputs?  If so, and if you can subsequently use only nans for bad values, you can do this:
```python
a = np.ma.masked_invalid(a).filled(np.nan)
```
Regardless of input, that will leave you with an ndarray with nothing but valid numbers and nans.  It handles inf as well as nan.